### PR TITLE
feat: Add kustomization for unspec

### PR DESCRIPTION
**Pull request by dharmendrak**

### CODE REVIEW

File: k8s/namespaces/unspec/ithc.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  name: unspec-service
+spec:
+  values:
+    java:
+      ingressHost: unspec-service-ithc.service.core-compute-ithc.internal
+      environment:
+        OIDC_ISSUER: https://forgerock-am.service.core-compute-idam-ithc.internal:8443/openam/oauth2/realms/root/realms/hmcts
+        TESTING_SUPPORT_ENABLED: true
+  global:
+    environment: ithc
```
Comment (adusumillipraveen) [inline k8s/namespaces/unspec/ithc.yaml:13]:
```suggestion
  
```

File: k8s/namespaces/unspec/unspec-service.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    path: stable/unspec-service
+  values:
+    java:
+      replicas: 2
+      useInterpodAntiAffinity: true
+      image: hmctspublic.azurecr.io/unspec/service:prod-d6cd6abc
+      aadIdentityName: unspec
+  global:
+    tenantId: "531ff96d-0ae9-462a-8d2d-bec7c0b42082"
+    enableKeyVaults: true
```
Comment (adusumillipraveen) [inline k8s/namespaces/unspec/unspec-service.yaml:26]:
```suggestion
```

File: k8s/namespaces/unspec/unspec-service.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+apiVersion: helm.fluxcd.io/v1
+kind: HelmRelease
+metadata:
+  name: unspec-service
+  namespace: unspec
+  annotations:
+    flux.weave.works/automated: "true"
+    flux.weave.works/tag.java: glob:prod-*
```
Comment (adusumillipraveen) [inline k8s/namespaces/unspec/unspec-service.yaml:8]:
```suggestion
```

File: k8s/namespaces/unspec/unspec-service.yaml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+spec:
+  releaseName: unspec-service
+  rollback:
+    enable: true
+    retry: true
```
Comment (adusumillipraveen) [inline k8s/namespaces/unspec/unspec-service.yaml:13]:
```suggestion
```

File: k8s/namespaces/unspec/unspec-service.yaml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  chart:
+    git: git@github.com:hmcts/hmcts-charts
+    ref: master
+    path: stable/unspec-service
+  values:
+    java:
+      replicas: 2
+      useInterpodAntiAffinity: true
+      image: hmctspublic.azurecr.io/unspec/service:prod-d6cd6abc
+      aadIdentityName: unspec
```
Comment (adusumillipraveen) [inline k8s/namespaces/unspec/unspec-service.yaml:23]: It is already in https://github.com/hmcts/hmcts-charts/blob/master/stable/unspec-service/values.yaml#L17-L18
```suggestion
    
```

File: k8s/namespaces/unspec/ithc.yaml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+apiVersion: helm.fluxcd.io/v1
+kind: HelmRelease
+metadata:
+  name: unspec-service
+spec:
+  values:
+    java:
+      ingressHost: unspec-service-ithc.service.core-compute-ithc.internal
```
Comment (adusumillipraveen) [inline k8s/namespaces/unspec/ithc.yaml:8]: Its already in https://github.com/hmcts/hmcts-charts/blob/master/stable/unspec-service/values.yaml#L17-L18
```suggestion
```

**Review by adusumillipraveen — CHANGES_REQUESTED** [review]

File: k8s/namespaces/unspec/ithc.yaml (hunk 6 above)
Comment (dharmendrak) [reply k8s/namespaces/unspec/ithc.yaml]: done

File: k8s/namespaces/unspec/demo.yaml [hunk 7]
```diff
@@ -0,0 +1,10 @@
+apiVersion: helm.fluxcd.io/v1
+kind: HelmRelease
+metadata:
+  name: unspec-service
+spec:
+  values:
+    java:
+      environment:
+        OIDC_ISSUER: https://forgerock-am.service.core-compute-idam-demo.internal:8443/openam/oauth2/realms/root/realms/hmcts
```
Comment (timja) [inline k8s/namespaces/unspec/demo.yaml:9]: this should use the environment field in the kustomization base and only be overridden where needed, in this PR that only appears to be in aat

Comment (timja) [reply k8s/namespaces/unspec/demo.yaml:9]: It's already set please remove and only override AAT unless i'm missing something
https://github.com/hmcts/unspec-service/blob/master/charts/unspec-service/values.yaml#L39

**Review by timja — CHANGES_REQUESTED** [review]

File: k8s/namespaces/unspec/kustomization.yaml [hunk 8]
```diff
@@ -4,3 +4,4 @@ namespace: unspec
 bases:
 - namespace.yaml
 # Warning : Adding a file here, adds to all environments to which you add your kustomization.
+- unspec-service.yaml
```
Comment (timja) [inline k8s/namespaces/unspec/kustomization.yaml:7]: this adds it to prod as well which I don't think is what you want, you should just be adding it to the environment specific ones for now

Comment (adusumillipraveen) [reply k8s/namespaces/unspec/kustomization.yaml:7]: It is not added to https://github.com/hmcts/cnp-flux-config/blob/master/k8s/prod/common-overlay/kustomization.yaml and being first app, i thought its fine.

**Review by adusumillipraveen — CHANGES_REQUESTED** [review]
`unspec-service` related files should go into `namespaces/unspec/unspec-service` directory

File: k8s/namespaces/unspec/kustomization.yaml (hunk 8 above)
Comment (timja) [reply k8s/namespaces/unspec/kustomization.yaml:7]: ah sure

File: k8s/namespaces/unspec/unspec-service/demo.yaml [hunk 9]
```diff
@@ -0,0 +1,9 @@
+apiVersion: helm.fluxcd.io/v1
```
Comment (timja) [inline k8s/namespaces/unspec/unspec-service/demo.yaml:1]: your patches need referencing in your environment specific overlay,
https://github.com/hmcts/cnp-flux-config/blob/master/k8s/prod/common-overlay/rpe/kustomization.yaml#L6-L7

**Review by timja — APPROVED** [review]
looks ok, I haven't looked terribly closely

**Review by adusumillipraveen — APPROVED** [review]

**Pull request merged**